vision/emergent_object/emergent_object_processing.py

```diff
@@ -0,0 +1,161 @@
+        A list of all candidate emergent objects. Required attributes:
+            latitude, longitude, bounding_box_area, image_path, confidence
+    odlcs: ODLC_Dict
+        A dictionary containing the locations of all discovered ODLCs
+
+    Returns
+    -------
+    emergent_object: BoundingBox
+        The candidate that has scored the highest, and so is probably the actual emergent
+        object
+    """
+
+    # Weights for each metric
+    category_weights: NDArray[Shape[4], Float64] = np.array([0.5, 0.25, 0.20, 0.05])
+
+    evaluations_list: list[tuple[float, float, float, float]] = []
+
+    subject_humanoid: BoundingBox
+    for subject_humanoid in humanoids:
+        # Get the distance to the closest standard object
+        min_odlc_distance: float = float("inf")
+        odlc: Location
+        for odlc in odlcs.values():
+            odlc_distance: float = geopy.distance.geodesic(
+                (odlc["latitude"], odlc["longitude"]),
+                (subject_humanoid.attributes["latitude"], subject_humanoid.attributes["longitude"]),
+            ).feet
+
+            min_odlc_distance = min(odlc_distance, min_odlc_distance)
+
+        # Get the distance to the closest emergent object that's not in the same image
+        min_humanoid_distance: float = float("inf")
+        humanoid: BoundingBox
+        for humanoid in humanoids:
+            # Don't check candidates that are in the same image
+            if subject_humanoid.attributes["image_path"] != humanoid.attributes["image_path"]:
+                humanoid_distance: float = geopy.distance.geodesic(
+                    (humanoid.attributes["latitude"], humanoid.attributes["longitude"]),
+                    (
+                        subject_humanoid.attributes["latitude"],
+                        subject_humanoid.attributes["longitude"],
+                    ),
+                ).feet
+
+                min_humanoid_distance = min(humanoid_distance, min_humanoid_distance)
+
+        # invert the humanoid distance to prioritize closest humanoids
+        min_humanoid_distance *= -1
+
+        evaluation: tuple[float, float, float, float] = (
+            min_odlc_distance,
+            min_humanoid_distance,
+            subject_humanoid.attributes["bounding_box_area"],
+            subject_humanoid.attributes["confidence"],
+        )
+
+        evaluations_list.append(evaluation)
+
+    evaluation_array: NDArray[Shape["*, 4"], Float64] = np.array(evaluations_list)
+
+    # Get the minimums of each category
+    evaluation_mins: NDArray[Shape["4"], Float64] = np.amin(evaluation_array, axis=0)
+
+    # Get the maximums of each category
+    evaluation_maxes: NDArray[Shape["4"], Float64] = np.amax(evaluation_array, axis=0)
+
+    # An error here could mean that all candidates were detected in the same image - This
+    #   is okay to ignore
+    with np.errstate(invalid="ignore"):
+        ranges: NDArray[Shape["4"], Float64] = evaluation_maxes - evaluation_mins
```

Comment:
Could you elaborate on this a bit? How does subtracting two arrays of floats have something to do with which image the boxes are in?

Reply:
Also, if possible, you should look for a way to do this that will not throw an error.